File: idea-plugin/src/main/java/ru/mipt/acsl/decode/idea/plugin/DecodeFileTypeFactory.java
package ru.mipt.acsl.decode.idea.plugin;

import com.intellij.openapi.fileTypes.FileTypeConsumer;
import com.intellij.openapi.fileTypes.FileTypeFactory;
import org.jetbrains.annotations.NotNull;
import ru.mipt.acsl.decode.parser.DecodeFileType$;

/**
 * @author Artem Shein
 */
public class DecodeFileTypeFactory extends FileTypeFactory
{
    @Override
    public void createFileTypes(@NotNull FileTypeConsumer consumer)
    {
        consumer.consume(DecodeFileType$.MODULE$.instance(), "decode");
    }
}

File: src/main/scala/ru/mipt/acsl/decode/idea/plugin/DecodeFileTypeFactory.scala
package ru.mipt.acsl.decode.idea.plugin

import com.intellij.openapi.fileTypes.FileTypeConsumer
import com.intellij.openapi.fileTypes.FileTypeFactory
import ru.mipt.acsl.decode.parser.DecodeFileType

/**
 * @author Artem Shein
 */
class DecodeFileTypeFactory extends FileTypeFactory {
    override def createFileTypes(consumer: FileTypeConsumer): Unit =
        consumer.consume(DecodeFileType, "decode")
}
